sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/TrigonometryTests.java

```diff
@@ -1118,4 +1118,200 @@ public void invalidInputs() {
                         "Supported form(s): 'SIN(<NUMERIC>)'"
         );
     }
+
+    // Tested on Apache Spark
+    @Test
+    public void testSec() {
+        this.qs("""
+                SELECT sec(0.6);
+                      sec
+                -----------------
+                 1.211628314512317
+                (1 row)
+                
+                SELECT sec(0);
+                      sec
+                -----------------
+                 1.0
+                (1 row)
+                
+                SELECT sec(PI);
+                      sec
+                -----------------
+                 -1.0
+                (1 row)
+                
+                SELECT sec(PI / 2);
+                      sec
+                -----------------
+                 16331239353195370
+                (1 row)
+                """
+        );
+    }
+
+    // Tested on Apache Spark
+    @Test
+    public void testCsc() {
+        this.qs("""
+                SELECT csc(0.6);
+                      csc
+                -----------------
+                 1.7710321966877254
+                (1 row)
+                
+                SELECT csc(0);
+                      csc
+                -----------------
+                 Infinity
+                (1 row)
+                
+                SELECT csc(PI);
+                      csc
+                -----------------
+                 8165619676597685
+                (1 row)
+                """
+        );
+    }
+
+    @Test
+    public void testHyperbolicFns() {
+        this.qs(
+                """
+                        SELECT sinh(1);
+                              sinh
+                        -----------------
+                         1.1752011936438014
+                        (1 row)
+                        
+                        SELECT asinh(1);
+                               asinh
+                        -------------------
+                         0.881373587019543
+                        (1 row)
+                        
+                        SELECT cosh(1);
+                               cosh
+                        ------------------
+                         1.5430806348152437
+                        (1 row)
+                        
+                        SELECT coth(0.6); -- not tested in Spark
```

Comment:
why not?

Reply:
It isn't available in Spark.

Reply:
I would have said "not available in Spark"